static/js/popovers.js

```diff
@@ -160,6 +161,11 @@ exports.toggle_actions_popover = function (element, id) {
             use_edit_icon = false;
             editability_menu_item = i18n.t("View source");
         }
+        if (message.backend_edit_arrived || message.backend_edit_arrived === undefined) {
```

Comment:
I would use `backend_edit_in_flight` here, so that we don't have to handle the awkward `undefined` case.  The normal situation for messages is to be **not** in-flight, so it's good to have the flag have the semantics that you have to explicitly set it to `true` to trigger exceptional behavior.

Reply:
To put it another way, I think all of our existing message booleans default to `false` when you create a message, so it's good to keep up with that pattern.

Reply:
And then for the template variable, you can simply say `show_edit_option = !message.backend_edit_in_flight`

Reply:
Resolved!